File: blink-macos/BlinkCore/Tests/BlinkCoreTests/MouseBehaviorScorerTests.swift
import Foundation
import Testing
@testable import BlinkCore

@Suite("MouseBehaviorScorer")
struct MouseBehaviorScorerTests {
    let scorer = MouseBehaviorScorer()

    @Test("No input = neutral score")
    func noInput() {
        let score = scorer.score(mouseEvents: [], keystrokeCount: 0, now: 1000)
        #expect(score == 0.5)
    }

    @Test("Keyboard only = high score")
    func keyboardOnly() {
        let score = scorer.score(mouseEvents: [], keystrokeCount: 50, now: 1000)
        #expect(score > 0.9, "Keyboard-dominant should be high score, got \(score)")
    }

    @Test("Mouse only = low score")
    func mouseOnly() {
        let events = (0..<20).map { MouseEvent(timestamp: 940 + Double($0) * 3, kind: .click) }
        let score = scorer.score(mouseEvents: events, keystrokeCount: 0, now: 1000)
        #expect(score < 0.2, "Mouse-only should be low score, got \(score)")
    }

    @Test("Heavy scrolling no typing = browsing")
    func heavyScrolling() {
        let events = (0..<25).map { MouseEvent(timestamp: 940 + Double($0) * 2, kind: .scroll(deltaY: 10)) }
        let score = scorer.score(mouseEvents: events, keystrokeCount: 2, now: 1000)
        #expect(score < 0.2, "Heavy scrolling with no typing = browsing, got \(score)")
    }
}

File: blink-windows/src/Blink.Core.Tests/Scorers/MouseBehaviorScorerTests.cs
using Blink.Core.Abstractions;
using Blink.Core.FlowDetection.SignalProcessors;

namespace Blink.Core.Tests.Scorers;

public class MouseBehaviorScorerTests
{
    private readonly MouseBehaviorScorer _scorer = new();

    [Fact] public void NoInput_NeutralScore() =>
        Assert.Equal(0.5, _scorer.Score([], 0, 1000));

    [Fact]
    public void KeyboardOnly_HighScore()
    {
        var score = _scorer.Score([], 50, 1000);
        Assert.True(score > 0.9, $"Keyboard-dominant should be high, got {score}");
    }

    [Fact]
    public void MouseOnly_LowScore()
    {
        var events = Enumerable.Range(0, 20)
            .Select(i => new MouseEvent(940 + i * 3, new MouseEventKind.Click())).ToList();
        var score = _scorer.Score(events, 0, 1000);
        Assert.True(score < 0.2, $"Mouse-only should be low, got {score}");
    }

    [Fact]
    public void HeavyScrolling_Browsing()
    {
        var events = Enumerable.Range(0, 25)
            .Select(i => new MouseEvent(940 + i * 2, new MouseEventKind.Scroll(10))).ToList();
        var score = _scorer.Score(events, 2, 1000);
        Assert.True(score < 0.2, $"Heavy scrolling = browsing, got {score}");
    }
}
